source/common/upstream/health_checker_impl.cc

```diff
@@ -336,7 +336,7 @@ Http::CodecClient::Type HttpHealthCheckerImpl::codecClientType(bool use_http2) {
 Http::CodecClient*
 ProdHttpHealthCheckerImpl::createCodecClient(Upstream::Host::CreateConnectionData& data) {
```

Comment:
I'd be inclined to snag the runtime value here too, so when we've flipped the default we can remove the old code path and always do full sanitization

Reply:
I think you did this in the wrong block, for the gRPC health checker (always H2) rather than than here where it can be 1/1

Other than that, LGTM.
@snowp any other thoughts on your end?

Reply:
@alyssawilk Good catch!  Just fixed that.